src/main/java/friday/Parser.java

```diff
@@ -0,0 +1,107 @@
+                    saveTasks(taskList.toString(), storage);
+                } else if (userInput.contains("find")) {
+                    String[] findInput = userInput.split(" ", 2);
+                    if (findInput.length < 2 || findInput[1].trim().isEmpty()) {
+                        ui.showFindError();
+                    }
+                    TaskList result = taskList.findTasks(findInput[1]);
+                    ui.findTasks(result);
+                } else if (userInput.contains("todo")) {
+                    String[] todoInput = userInput.split(" ", 2);
+                    if (todoInput.length < 2 || todoInput[1].trim().isEmpty()) {
+                        ui.showTodoError();
+                    }
+                    Todo todo = new Todo(todoInput[1]);
+                    ui.addTask(todo);
+                    taskList.add(todo);
+                    ui.taskListMessage(taskList);
+                    saveTasks(taskList.toString(), storage);
+
+                } else if (userInput.contains("deadline")) {
+                    String[] commandAndDetails = userInput.split(" ", 2);
+                    if (commandAndDetails.length < 2 || !userInput.contains("/by")) {
+                        ui.showDeadlineFormatError();
+                    }
+                    String[] taskAndDate = commandAndDetails[1].split(" /by ", 2);
+                    if (taskAndDate.length < 2) {
+                        ui.showDeadlineMissingError();
+                    }
+                    String taskDescription = taskAndDate[0];
+                    String deadlineDate = taskAndDate[1];
+                    Deadline deadline = new Deadline(taskDescription, deadlineDate);
+                    ui.addTask(deadline);
+                    taskList.add(deadline);
+                    ui.taskListMessage(taskList);
+                    saveTasks(taskList.toString(), storage);
+                } else if (userInput.contains("event")) {
+                    String[] commandAndDetails = userInput.split(" ", 2);
+                    if (commandAndDetails.length < 2 || !userInput.contains("/from") || !userInput.contains("/to")) {
+                        ui.showEventFormatError();
+                    }
+                    String[] taskAndTimes = commandAndDetails[1].split(" /from | /to ", 3);
+                    if (taskAndTimes.length < 3) {
+                        ui.showEventMissingError();
+                    }
+                    String taskDescription = taskAndTimes[0];
+                    String startTime = taskAndTimes[1];
+                    String endTime = taskAndTimes[2];
+                    Event event = new Event(taskDescription, startTime, endTime);
+                    ui.addTask(event);
+                    taskList.add(event);
+                    ui.taskListMessage(taskList);
+                    saveTasks(taskList.toString(), storage);
+                } else {
+                    ui.showBadInputError();
+                }
```

Comment:
Generally for these if-else statements, maybe there could be empty lines separating some groups of related statements? Not too important but it would improve readability.